### DISCUSSION TITLE
Request for Compatibility with Python 3.12 - 3.13

### DISCUSSION THREAD

**Discussion by ArnaudPolitico** [open-webui/open-webui#7273] [Ideas]
Hi there, 

First, I’d like to thank you for the incredible work you’ve done on OpenWebUI. It’s an essential tool, and I’ve greatly appreciated its capabilities and performance.

I’m currently using OpenWebUI on Windows, and I’ve encountered an issue related to Python version compatibility. As of now, OpenWebUI supports Python 3.11, but Python.org has recently stopped providing binary installers for this version on Windows. According to their release notes for Python 3.11.10 (released on September 7, 2024), Python 3.11 has entered the “security fixes only” stage, and all future releases will be source-only until the end of its lifecycle in October 2027.

The decision not to provide binary installers for Python 3.11 appears to be aligned with their lifecycle policy, which focuses on encouraging users to migrate to newer, actively supported versions, such as Python 3.12. However, this poses challenges for OpenWebUI users on Windows, as building Python from source is often impractical for non-developers.

I also want to mention that I strongly prefer using Python directly over Docker because I find Docker to be significantly heavier in terms of performance and resource usage. A Python-native solution has consistently proven to be more efficient and aligns better with my usage preferences.

Given the above, I would like to ask:

Are there plans to update OpenWebUI to support Python 3.12 in the near future?
If not, are there any workarounds or recommendations for Windows users facing the absence of binary installers for Python 3.11?
Updating OpenWebUI to support Python 3.12 would help ensure compatibility with actively supported Python versions and maintain accessibility for Windows users.

Thank you for considering this request, and I’d be happy to assist with testing or providing feedback if needed.

Best regards,

**Comment by tjbck** [comment]
Not on the immediate roadmap, however, I'd appreciate community contributions in this area!

**Reply by ArnaudPolitico** [reply]
It's annoying because, for example, I don't have the knowledge to compile the source code, and as a result, I’m using an obscure version 3.11.0 of Python I found on the web...

**Reply by hilam** [reply]
Greetings to everyone.

@tjbck Could you mention what you think would have to change?

Because of the intensive use of f-strings in the open-webui development environment (tools, functions, pipelines), and the improvement in nested f-strings promoted by this Python version, I am interested in helping with this task, if I can measure the required effort.

I really appreciate your efforts and those of all the contributors to this wonderful tool, and I will be happy to contribute whenever I can.

Hugs!
 
(PS: I'm a Brazilian Python dev, and contributor to translating the documentation into Brazilian Portuguese)
